src/constants/types.rs

```diff
@@ -513,6 +514,50 @@ impl ResourceType {
 
 js_deserializable!(ResourceType);
 
+/// Translates market resource types which can include both `RESOURCE_*`
+/// and `INTERSHARD_RESOURCES` constants.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum MarketResourceType {
+    Resource(ResourceType),
+    IntershardResource(IntershardResourceType),
+}
+
+impl<'de> Deserialize<'de> for MarketResourceType {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: Cow<'de, str> = Cow::deserialize(d)?;
+
+        // first try to deserialize as a ResourceType
+        match ResourceType::from_str(&s) {
```

Comment:
You could probably have this look a bit nicer using the `Result` methods, but that code aesthetic.

Reply:
Updated, I tried to make it not too big of a mess but the formatter had other ideas ;)